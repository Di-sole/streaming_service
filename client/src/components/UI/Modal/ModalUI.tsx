import { ReactNode, useEffect } from 'react';
import styles from './ModalUI.module.scss';
import Image from 'next/image';
import closeIcon from '../../../../public/icons/close.svg';
import { useRouter } from 'next/router';
import TransparentButton from '../buttons/TransparentButton/TransparentButton';
interface ModalUIProps {
    children: ReactNode;
    className?: string;
    close?: boolean;
}

const ModalUI = ({ children, className, close }: ModalUIProps) => {
    const location = useRouter();
    const backPath = location.asPath.replace(
        /(\?ivi_search)|(\?sign-in)|(\?sign-up)|(\?authorized)|(\?trailer)|(\?more)|(\?validation=)|(\?validation)/,
        '',
    );

    const body = typeof document !== 'undefined' ? document?.querySelector('body') : true;

    useEffect(() => {
        body !== true && body?.classList.add('modal-active');
    }, [location.asPath]);

    const closeHandler = () => {
        body !== true && body?.classList.remove('modal-active');
        location.push(backPath);
    };

    useEffect(() => {
        close && body !== true && body?.classList.remove('modal-active');
        close && location.asPath.includes('?authorized' || '?sign-in' || '?sign-up' || '?validation') && closeHandler();

        body === true && closeHandler();
    }, [close]);

    return (
        <div className={[styles.container, className].join(' ')}>
            {children}

            <TransparentButton textColor="bright" className={styles.container__button} onClick={closeHandler}>
                <Image className={styles.container__image} src={closeIcon} alt="закрыть" />
            </TransparentButton>
        </div>
    );
};

export default ModalUI;
