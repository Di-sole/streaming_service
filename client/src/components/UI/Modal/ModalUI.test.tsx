import { render, screen } from '@testing-library/react';
import ModalUI from './ModalUI';

jest.mock('next/router', () => ({
    useRouter() {
        return {
            pathname: '',
            asPath: '',
        };
    },
}));

describe('MODALUI SNAPSHOTS TESTS', () => {
    test('ModalUI should not be changed', () => {
        render(<ModalUI>Я модалка</ModalUI>);
        const modal = screen.getByText('Я модалка');
        expect(modal).toMatchSnapshot();
    });
});
