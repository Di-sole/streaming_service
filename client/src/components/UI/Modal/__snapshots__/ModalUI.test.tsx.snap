// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalUI SNAPSHOTS TESTS ModalUI should not be changed 1`] = `
<div
  class="container "
>
  Я модалка
  <button
    class="button button_bright container__button"
    type="button"
  >
    <img
      alt="закрыть"
      class="container__image"
      data-nimg="1"
      decoding="async"
      height="40"
      loading="lazy"
      src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
      srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
      style="color: transparent;"
      width="40"
    />
  </button>
</div>
`;
